pkg/operator/ceph/config/config.go

```diff
@@ -0,0 +1,199 @@
+// not exist, a new empty section is created and returned for the caller's use.
+func (c *Config) Section(name string) *Section {
+	s, ok := c.sections[name]
+	if !ok {
+		s = &Section{
+			configs:     make(map[string]string),
+			configOrder: []string{},
+		}
+		c.sections[name] = s
+		c.sectionOrder = append(c.sectionOrder, name)
+	}
+	return s
+}
+
+// Set sets the Ceph config key to the given value. Set accepts Ceph keys in both
+// space-separated-words and underscore-separated-words formats. Set modifies the section in-place
+// and also returns the same modified-in-place Section to give flexibility of usage.
+//
+// NOTE that if a config item in a map should have quotes around it, they will have to be added to
+// the string explicitly. e.g., a Value of " test " will be interpreted as as 'Key = test', whereas
+// a Value of "\" test \"" will be interpreted as 'Key = " test "'
+func (s *Section) Set(key, value string) *Section {
+	key = normalizeKey(key)
+	_, ok := s.configs[key]
+	if !ok {
+		s.configOrder = append(s.configOrder, key)
+	}
+	s.configs[key] = value
+	return s
+}
+
+// Merge modifies the Section in-place adding values from the overrides Section and overriding
+// existing values with the new value in overrides.
+func (s *Section) Merge(overrides *Section) {
+	for k, v := range overrides.configs {
+		s.Set(k, v)
+	}
+}
+
+// Merge modifies the Config in-place, adding Sections from the overrides config, and overriding
+// existing Sections with Section values from overrides.
+func (c *Config) Merge(overrides *Config) {
+	for hdr, sec := range overrides.sections {
+		c.Section(hdr).Merge(sec)
+	}
+}
+
+// FileText converts the Ceph Config to a string formatted to look like the text of a Ceph config
+// file. If errors are encountered, the error is returned with an empty string.
+// Double newlines after headers and at the end of a section; single newline betw header and section
+func (c *Config) FileText() (string, error) {
+	f := ini.Empty()
+	for _, hdr := range c.sectionOrder {
+		sec, ok := c.sections[hdr]
+		if !ok {
+			return "", fmt.Errorf("Error: section header %s not found: %+v", hdr, c)
+		}
+		s, err := f.NewSection(hdr)
+		if err != nil {
+			return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+		}
+		for _, k := range sec.configOrder {
+			v, ok := sec.configs[k]
+			if !ok {
+				return "", fmt.Errorf("Error: config %s not found in section %s: %+v", k, hdr, sec)
+			}
+			if _, err := s.NewKey(k, v); err != nil {
+				return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+			}
+		}
+	}
+	b := new(bytes.Buffer)
+	if _, err := f.WriteTo(b); err != nil {
+		return "", fmt.Errorf("Error converting config to ini file text: %+v. %+v", c, err)
+	}
+	return b.String(), nil
+}
+
+// NewFlag returns the key-value pair in the format of a Ceph command line-compatible flag.
+func NewFlag(key, value string) string {
```

Comment:
Have we confirmed this is the compatible way to convert all ceph config settings to cmd line args?

Reply:
looks correct to me

the only oddity is that for boolean options you can either do ``--foo`` and ``--no-foo`` or ``--foo=true`` and ``--foo=false``.  Looks like this method will do the latter.

Reply:
Ceph has special logic for what a boolean flag looks like in the config file `key = true/false` versus what it looks like on the commandline `--key/--no-key`?